tools/analyze_logs.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+"""Modified from https://github.com/open-
+mmlab/mmdetection/blob/master/tools/analysis_tools/analyze_logs.py."""
+import argparse
+import json
+from collections import defaultdict
+
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+
+def cal_train_time(log_dicts, args):
+    for i, log_dict in enumerate(log_dicts):
+        print(f'{"-" * 5}Analyze train time of {args.json_logs[i]}{"-" * 5}')
+        all_times = []
+        for epoch in log_dict.keys():
+            if args.include_outliers:
+                all_times.append(log_dict[epoch]['time'])
+            else:
+                all_times.append(log_dict[epoch]['time'][1:])
+        all_times = np.array(all_times)
+        epoch_ave_time = all_times.mean(-1)
+        slowest_epoch = epoch_ave_time.argmax()
+        fastest_epoch = epoch_ave_time.argmin()
+        std_over_epoch = epoch_ave_time.std()
+        print(f'slowest epoch {slowest_epoch + 1}, '
+              f'average time is {epoch_ave_time[slowest_epoch]:.4f}')
+        print(f'fastest epoch {fastest_epoch + 1}, '
+              f'average time is {epoch_ave_time[fastest_epoch]:.4f}')
+        print(f'time std over epochs is {std_over_epoch:.4f}')
+        print(f'average iter time: {np.mean(all_times):.4f} s/iter')
+        print()
+
+
+def plot_curve(log_dicts, args):
+    if args.backend is not None:
+        plt.switch_backend(args.backend)
+    sns.set_style(args.style)
+    # if legend is None, use {filename}_{key} as legend
+    legend = args.legend
+    if legend is None:
+        legend = []
+        for json_log in args.json_logs:
+            for metric in args.keys:
+                legend.append(f'{json_log}_{metric}')
+    assert len(legend) == (len(args.json_logs) * len(args.keys))
+    metrics = args.keys
+
+    num_metrics = len(metrics)
+    for i, log_dict in enumerate(log_dicts):
+        epochs = list(log_dict.keys())
+        for j, metric in enumerate(metrics):
+            print(f'Plot curve of {args.json_logs[i]}, metric is {metric}')
+
+            if 'hmean' in metric or '0_' in metric:
```

Comment:
```suggestion
            if metric in ['hmean', 'word_acc', 'word_acc_ignore_case', 
                'word_acc_ignore_case_symbol', 'char_recall', 'char_precision',
                '1-N.E.D', `macro_f1`]
```
It doesn't cover all the cases yet. For text recognition models, we usually evaluate it on 6 or even more datasets and therefore, the leading index is not always 0. I'd suggest the above change, though it need some linting afterward

Reply:
Thanks! It's necessary for us to adapt to all situations. After changing, I'm wondering if `hmean-iou:hmean` or `0_word_acc` can be used to plot? They are not in the list. In addition, the style `macro_f1` in list may went wrong.

Reply:
Thanks for pointing it out, I made a mistake here. It should look like the following:
```python
epoch_based_metrics = ['hmean', 'word_acc', 'word_acc_ignore_case', 
	                'word_acc_ignore_case_symbol', 'char_recall', 'char_precision',
	                '1-N.E.D', `macro_f1`]
if any(metric in m for m in epoch_based_metrics):
```
Not sure if we can put it in a more elegant way. BTW, `macro_f1` is right, see an example https://download.openmmlab.com/mmocr/kie/sdmgr/20210520_132236.log.json

Reply:
It's nearly perfect! I'm sorry I didn't express it clearly, I meant the `marco_f1` should be a string end with `' '` instead of ` `` `, it might be a typo. Don't mind, I fixed it in the latest commit. Thanks again!